### PR TITLE
fix(SD): add a display timer on status page

### LINKED ISSUE #8046: STS page on ecam
### Aircraft Version

Development

### Build info

```json
{
    "built": "2023-05-23T08:10:56+00:00",
    "ref": "refs/heads/experimental",
    "sha": "7ac70bfd908988bfdc68879b16a169d26bf0c72c",
    "actor": "Saschl",
    "event_name": "manual",
    "pretty_release_name": "experimental:7ac70bfd",
    "version": "v0.10.0-exp.7ac70bf"
}
```

### Describe the bug

Well, when you push STS button on ECAM control panel, then status page is shown, Thats ok. But when status is normal, which is inidicated with word normal on middle of ecam, then page returns itself to previous mode, whichever that was. Normally if there are any faults or messages, it would stay, but in FBW case it stays even if itas normaly, And thats not correct, if its normaly it should roll back to one before in 4 second, lets say 4. Kind regards

### Expected behavior

explained before

### Steps to reproduce

STS need to roll back if status is normal to page we were before, or at least to first one. And when in cruise, to cruise page

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

addict2flight

### PR DESCRIPTION
**Pull request by pluce**
Fixes #8046 

## Summary of Changes
The Status Page page of lower ECAM / System Display should appear only 3s if no failures are to be shown. This PR adds this timer before coming back to previous page.
Also replaces hardcoded values by constants to improve readability in the PageContainer component.

## References
[FCOM](http://repo.poltekbangsby.ac.id/571/1/A318_A319_A320_A321%20%28%20PDFDrive%20%29.pdf) (page 1190)
STS_button

## Additional context
First PR 👀 

## Testing instructions
Should not change behaviour of lower ECAM beside returning to previous page after 3s if no failure. I couldn't make Status page to be filled by anything, but after a discussion on Discord it may be normal.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by pluce** [comment]
- [ ] Change the constants list for pages to an Enum

**Comment by 2hwk** [comment]
Lint is failing, please fix
https://github.com/flybywiresim/a32nx/actions/runs/5125901688/jobs/9223580234?pr=8048
image

File: fbw-a32nx/src/systems/instruments/src/SD/PagesContainer.tsx [hunk 1]
```diff
@@ -61,11 +87,31 @@ export const PagesContainer = () => {
...
+        if(currentPage !== SdPages.Status) {
+            return
+        }
+
+        if (isStatusPageEmpty) {
+            if(stsPressedTimer > 0) {
+                setStsPressedTimer((prev) => prev - deltaTime / 1000);
+                setPageWhenUnselected(SdPages.Status);
+            } else {
+                SimVar.SetSimVarValue(`L:A32NX_ECAM_SD_CURRENT_PAGE_INDEX`, "number", stsPrevPage);
```
Comment (2hwk) [inline fbw-a32nx/src/systems/instruments/src/SD/PagesContainer.tsx:110]:
```suggestion
                SimVar.SetSimVarValue(`L:A32NX_ECAM_SD_CURRENT_PAGE_INDEX`, 'number', stsPrevPage);
```

**Review by 2hwk — CHANGES_REQUESTED** [review]

File: fbw-a32nx/src/systems/instruments/src/SD/PagesContainer.tsx [hunk 2]
```diff
@@ -61,11 +87,31 @@ export const PagesContainer = () => {
...
+            }
+        } else {
+            setPageWhenUnselected(SdPages.Status);
+            setStsPressedTimer(STS_DISPLAY_TIMER_DURATION)
```
Comment (2hwk) [inline fbw-a32nx/src/systems/instruments/src/SD/PagesContainer.tsx:114]:
```suggestion
            setStsPressedTimer(STS_DISPLAY_TIMER_DURATION);
```

File: fbw-a32nx/src/systems/instruments/src/SD/PagesContainer.tsx [hunk 3]
```diff
@@ -143,36 +192,41 @@ export const PagesContainer = () => {
...
             default:
                 // Sometimes happens when loading in, in which case we have to initialise pageNameWhenUnselected here.
-                setPageWhenUnselected(8);
+                setPageWhenUnselected(SdPages.Door);
                 break;
             }
 
-            if (failPage !== -1) {
+            if (newPage === SdPages.Status && currentPage !== newPage) {
+                setStsPressedTimer(STS_DISPLAY_TIMER_DURATION)
```
Comment (2hwk) [inline fbw-a32nx/src/systems/instruments/src/SD/PagesContainer.tsx:214]:
```suggestion
                setStsPressedTimer(STS_DISPLAY_TIMER_DURATION);
```

File: fbw-a32nx/src/systems/instruments/src/SD/PagesContainer.tsx [hunk 4]
```diff
@@ -16,22 +16,48 @@ import { FctlPage } from './Pages/Fctl/Fctl';
...
     const [fwcFlightPhase] = useSimVar('L:A32NX_FWC_FLIGHT_PHASE', 'Enum', 500);
-    const [crzCondTimer, setCrzCondTimer] = useState(60);
-    const [ecamFCTLTimer, setEcamFCTLTimer] = useState(20);
+    const [crzCondTimer, setCrzCondTimer] = useState(CRZ_CONDITION_TIMER_DURATION);
+    const [ecamFCTLTimer, setEcamFCTLTimer] = useState(FCTL_CONDITION_TIMER_DURATION);
     const [mainEngineStarterOffTimer, setMainEngineStarterOffTimer] = useState(-1);
     const [apuAboveThresholdTimer, setApuAboveThresholdTimer] = useState(-1);
+    const [stsPressedTimer, setStsPressedTimer] = useState(-1);
+    const [stsPrevPage, setStsPrevPage] = useState(-1);
+    const [normalStatusLine] = useSimVar(`L:A32NX_STATUS_LEFT_LINE_8`, 'number', 100);
```
Comment (2hwk) [inline fbw-a32nx/src/systems/instruments/src/SD/PagesContainer.tsx:60]:
```suggestion
    const [normalStatusLine] = useSimVar('L:A32NX_STATUS_LEFT_LINE_8', 'number', 100);
```

**Review by 2hwk — CHANGES_REQUESTED** [review]

**Comment by pluce** [comment]
should be good @2hwk thanks !

**Review by Benjozork — APPROVED** [review]

**Review by 2hwk — APPROVED** [review]

**Comment by Mack674** [comment]
Quality Assurance Tester Report

Discord: Mack674#2274
Object of testing: #8048
Tier of Testing: 1
Date: 30/06/2023

Testing Process:
- Spawned C&D at the gate
- Turned on Batteries 1&2 and external power
- Checked behaviour of STS ecam page

Negatives: N/A       

Testing Results: Passed 

Conclusions: Works as intended. Great job!

Media:

https://github.com/flybywiresim/a32nx/assets/62873705/634234cd-3153-4d68-8e95-4f91bf128f4c

**Pull request merged**